pkg/dnsmasq/nanny.go

```diff
@@ -69,6 +70,11 @@ func (n *Nanny) Configure(args []string, config *config.Config) {
 
 	for domain, serverList := range config.StubDomains {
 		for _, server := range serverList {
+			if isIP := (net.ParseIP(server) != nil); !isIP {
+				if IPs, err := net.LookupIP(server); err == nil && len(IPs) > 0 {
```

Comment:
Does this lookup work for the initial setup --- when dnsmasq has not been started?
https://github.com/kubernetes/dns/blob/a038db946ea2aeefe848116325c96644915f503d/pkg/dnsmasq/nanny.go#L179-L182